icu4j/main/classes/core/src/com/ibm/icu/impl/number/DecimalQuantity_AbstractBCD.java

```diff
@@ -216,6 +222,15 @@ public void adjustMagnitude(int delta) {
         }
     }
 
+    @Override
+    public int getOrigMagnitude() throws ArithmeticException {
+        DecimalQuantity_AbstractBCD copy = (DecimalQuantity_AbstractBCD) createCopy();
```

Comment:
Why do you need to make a copy here?  The arithmetic should be simple.  I think it's just the difference (or sum?) of `origDelta` and `getMagnitude()` on `this`.

Reply:
Done.  Using exponent field instead.